src/PPN-microbench/memory.cpp

```diff
@@ -0,0 +1,81 @@
+#include <PPN-microbench/memory.hpp>
+
+
+// Function to generate a random index within a given threshold
+inline u64 random_index(u64 threshold) { return rand() % threshold; }
+
+// Function to measure cache/memory latency using pointer chasing
+double measure_latency(u64 size, u64 nbIterations) {
+    std::vector<void *> memblock(size);
+
+    // Initialize the memory block with pointers to itself
+    for (u64 i = 0; i < size; ++i) {
+        memblock[i] = &memblock[i];
+    }
+
+    // Shuffle the pointers to create a random access pattern
+    for (u64 i = size - 1; i > 0; --i) {
+        u64 j = random_index(i + 1);
+        std::swap(memblock[i], memblock[j]);
+    }
+
+    double total_latency = 0.0;
+
+    // Perform the latency measurement 10 times for better accuracy
+    for (int run = 0; run < 10; ++run) {
```

Comment:
Maybe add a couple of warmup runs so the cpu has time to speed up before you save results